src/gsi/read_gmi.f90

```diff
@@ -528,6 +528,7 @@ subroutine read_gmi(mype,val_gmi,ithin,rmesh,jsatid,gstime,&
         flgch = 0
 
         iobs=iobs+1
+        if(iobs>maxobs) exit
```

Comment:
While this works, a different approach is used in `read_atms.f90`.  This routine uses `iob` in the same as as `read_gmi.f90` uses `iobs`.   `iob` is included in the logic which control looping (reading) through the atms bufr file
```
It would be nice to use a consistent approach throughout GSI observation readers.
!    Loop to read bufr file
     read_subset: do while(ireadmg(lnbufr,subset,idate)>=0 .AND. iob < maxobs)

        read_loop: do while (ireadsb(lnbufr)==0 .and. iob < maxobs)
```

Reply:
Before we added this line, we noticed there are 19 lines as "if(ndata>maxobs)" and "if(iobs>maxobs)" in 13 subroutine files, so we chose to add one line "if(iobs>maxobs) exit" or "if(ndata>maxobs) exit" to subroutines instead of changing exist lines.

Reply:
Thanks for explaining your logic.  Ensuring consistency is challenging without a code manager.